### PR TITLE
Fix #2555: PUT APIs should only allow admin and owners to update a resource in secure env

### LINKED ISSUE #2555: PUT APIs should only allow admin and owners to update a resource in secure env
**Affected module**
Does it impact the UI, backend or Ingestion Framework?
Backend

**Describe the bug**
Put APIs should not allow non-admin or non-owners to update any resource.

### PR DESCRIPTION
**Pull request by vivekratnavel**
### Describe your changes :
I added check for admin or owner permissions before updating any resource via PUT API.

#
### Type of change :

- [x] Bug fix

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [ ] I have tagged my reviewers below.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

### CODE REVIEW

File: ingestion/src/metadata/ingestion/sink/metadata_rest.py [hunk 1]
```diff
@@ -145,7 +145,7 @@ def write_tables(self, db_and_table: OMetaDatabaseAndTable):
                 tableType=db_and_table.table.tableType,
                 columns=db_and_table.table.columns,
                 description=db_and_table.table.description,
-                database=db.id,
+                database=db,
```
Comment (harshach) [inline ingestion/src/metadata/ingestion/sink/metadata_rest.py:148]: @vivekratnavel make sure this is a entity reference

Comment (vivekratnavel) [reply ingestion/src/metadata/ingestion/sink/metadata_rest.py]: Good catch. Fixed it in the latest patch and verified that sample data ingestion works.

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/ChartRepository.java [hunk 2]
```diff
@@ -126,9 +128,11 @@ private EntityReference getService(Chart chart) throws IOException, ParseExcepti
 
   public static class ChartEntityInterface implements EntityInterface<Chart> {
     private final Chart entity;
+    private final String fqn;
```
Comment (harshach) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/ChartRepository.java:131]: why do we want to keep this around as member?

Comment (vivekratnavel) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/ChartRepository.java]: Fixed it.

Comment (harshach) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/ChartRepository.java]: @vivekratnavel not sure what was the fix. But my question was around why we need to have "fqn" as the member variable why not make that call and get it on demand?

Comment (vivekratnavel) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/ChartRepository.java]: Yes, that's what I am doing with the latest patch. Getting the fqn on demand - https://github.com/open-metadata/OpenMetadata/pull/2608/files#diff-a981d13766f798ba878ac3c8867295329e36539829b1b5f6cad3b9fa28b81aa3R163

**Review by harshach — APPROVED** [review]

**Pull request merged**